src/CoreBundle/private/library/classes/dbobjects/TCMSImageEndpoint.class.php

```diff
@@ -733,11 +733,14 @@ public static function imagecopymerge_alpha($dst_im, $src_im, $dst_x, $dst_y, $s
      */
     public function GetRelativeURL()
     {
-        if (!$this->_isThumbnail) {
-            $sImageURL = $this->GetImageUrlPathPrefix().$this->aData['path'];
-        } else {
-            $sImageURL = URL_MEDIA_LIBRARY_THUMBS_PATH.$this->GetThumbPathExtension().'/'.$this->aData['path'];
+        /** @var string|null $dataPath */
+        $dataPath = $this->aData['path'] ?? '';
+
+        $sImageURL = URL_MEDIA_LIBRARY_THUMBS_PATH.$this->GetThumbPathExtension().'/'.$dataPath;
+        if (false === $this->_isThumbnail) {
```

Comment:
This changes the original if-else logic.

At least if this if is true then the previsous line was executed superfluously.

Reply:
true, you'r aright. Else is back.